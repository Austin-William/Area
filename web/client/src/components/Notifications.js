import React from "react";
import { url } from "../global/variables";
import "../styles/Notifications.css";
import { IoIosClose } from 'react-icons/io';
import axios from 'axios';

function Notifications(props) {

    const handleDelete = () => {
        let newNotifications = props.currentUserData.notifications;
        newNotifications.splice(props.index, 1);
        axios.put(url + '/api/user/' + props.currentUserData.id, {
            "notifications": newNotifications
        });
    }

    return (
        <div className="Notifications">
            <div className="Notifications-container">
                <button className="Notifications-close-button" type="button" onClick={handleDelete}>
                    <IoIosClose className="Widget-icon" />
                </button>
                <div className="Notifications-content">
                    {props.content}
                </div>
            </div>
        </div>
    );
}

export default Notifications;