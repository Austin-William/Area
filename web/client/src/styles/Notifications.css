.Notifications {
    display: flex;
    flex-direction: row;
    flex-wrap: wrap;
    justify-content: space-between;
    background-color: whitesmoke;
    border-radius: 0.5em;
    box-shadow: 0 0 0.5em 0.5em rgba(196, 196, 196, 0.1);
    margin-bottom: 1em;
}

.Notifications:hover {
    box-shadow: 0 0 0.5em 0.5em rgba(196, 196, 196, 0.479);
}

.Notifications-container {
    display: flex;
    flex-direction: row;
    justify-content: space-between;
    align-items: center;
    padding: 0.5em;
}

.Notifications-content {
    display: flex;
    align-items: center;
    justify-content: flex-end;
    height: fit-content;
    text-align: justify;
}

.Notifications-close-button {
    background-color: transparent;
    border: none;
    cursor: pointer;
    outline: none;
    padding: 0;
    border-radius: 0.5em;
    transition: all 0.2s ease-in-out;
    height: 100%;
    margin-right: 1em;
}

.Notifications-close-button:hover {
    background-color: red;
}

.Notifications-icon {
    width: 1.5em;
    height: 1.5em;
    margin-right: 0.5em;
}