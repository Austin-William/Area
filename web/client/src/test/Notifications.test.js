import React from "react";
import renderer from "react-test-renderer";
import Notifications from "../components/Notifications";

test("Components : Notifications renders correctly", () => {
  const tree = renderer.create(<Notifications />).toJSON();
  expect(tree).toMatchSnapshot();
});

test("Components : Notifications renders correctly", () => {
  var userData = {};
  const tree = renderer.create(<Notifications currentUserData={userData} />).toJSON();
  expect(tree).toMatchSnapshot();
});

test("Components : Notifications renders correctly", () => {
  const tree = renderer.create(<Notifications content="Random text" />).toJSON();
  expect(tree).toMatchSnapshot();
});

test("Components : Notifications renders correctly", () => {
  const tree = renderer.create(<Notifications index={2} />).toJSON();
  expect(tree).toMatchSnapshot();
});

test("Components : Notifications renders correctly", () => {
  const tree = renderer.create(<Notifications id="gjdsbfghsdihvsdv" />).toJSON();
  expect(tree).toMatchSnapshot();
});
