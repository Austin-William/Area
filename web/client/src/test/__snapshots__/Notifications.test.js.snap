// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Components : Notifications renders correctly 1`] = `
<div
  className="Notifications"
>
  <div
    className="Notifications-container"
  >
    <button
      className="Notifications-close-button"
      onClick={[Function]}
      type="button"
    >
      <svg
        className="Widget-icon"
        fill="currentColor"
        height="1em"
        stroke="currentColor"
        strokeWidth="0"
        style={
          Object {
            "color": undefined,
          }
        }
        viewBox="0 0 512 512"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M278.6 256l68.2-68.2c6.2-6.2 6.2-16.4 0-22.6-6.2-6.2-16.4-6.2-22.6 0L256 233.4l-68.2-68.2c-6.2-6.2-16.4-6.2-22.6 0-3.1 3.1-4.7 7.2-4.7 11.3 0 4.1 1.6 8.2 4.7 11.3l68.2 68.2-68.2 68.2c-3.1 3.1-4.7 7.2-4.7 11.3 0 4.1 1.6 8.2 4.7 11.3 6.2 6.2 16.4 6.2 22.6 0l68.2-68.2 68.2 68.2c6.2 6.2 16.4 6.2 22.6 0 6.2-6.2 6.2-16.4 0-22.6L278.6 256z"
        />
      </svg>
    </button>
    <div
      className="Notifications-content"
    />
  </div>
</div>
`;

exports[`Components : Notifications renders correctly 2`] = `
<div
  className="Notifications"
>
  <div
    className="Notifications-container"
  >
    <button
      className="Notifications-close-button"
      onClick={[Function]}
      type="button"
    >
      <svg
        className="Widget-icon"
        fill="currentColor"
        height="1em"
        stroke="currentColor"
        strokeWidth="0"
        style={
          Object {
            "color": undefined,
          }
        }
        viewBox="0 0 512 512"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M278.6 256l68.2-68.2c6.2-6.2 6.2-16.4 0-22.6-6.2-6.2-16.4-6.2-22.6 0L256 233.4l-68.2-68.2c-6.2-6.2-16.4-6.2-22.6 0-3.1 3.1-4.7 7.2-4.7 11.3 0 4.1 1.6 8.2 4.7 11.3l68.2 68.2-68.2 68.2c-3.1 3.1-4.7 7.2-4.7 11.3 0 4.1 1.6 8.2 4.7 11.3 6.2 6.2 16.4 6.2 22.6 0l68.2-68.2 68.2 68.2c6.2 6.2 16.4 6.2 22.6 0 6.2-6.2 6.2-16.4 0-22.6L278.6 256z"
        />
      </svg>
    </button>
    <div
      className="Notifications-content"
    />
  </div>
</div>
`;

exports[`Components : Notifications renders correctly 3`] = `
<div
  className="Notifications"
>
  <div
    className="Notifications-container"
  >
    <button
      className="Notifications-close-button"
      onClick={[Function]}
      type="button"
    >
      <svg
        className="Widget-icon"
        fill="currentColor"
        height="1em"
        stroke="currentColor"
        strokeWidth="0"
        style={
          Object {
            "color": undefined,
          }
        }
        viewBox="0 0 512 512"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M278.6 256l68.2-68.2c6.2-6.2 6.2-16.4 0-22.6-6.2-6.2-16.4-6.2-22.6 0L256 233.4l-68.2-68.2c-6.2-6.2-16.4-6.2-22.6 0-3.1 3.1-4.7 7.2-4.7 11.3 0 4.1 1.6 8.2 4.7 11.3l68.2 68.2-68.2 68.2c-3.1 3.1-4.7 7.2-4.7 11.3 0 4.1 1.6 8.2 4.7 11.3 6.2 6.2 16.4 6.2 22.6 0l68.2-68.2 68.2 68.2c6.2 6.2 16.4 6.2 22.6 0 6.2-6.2 6.2-16.4 0-22.6L278.6 256z"
        />
      </svg>
    </button>
    <div
      className="Notifications-content"
    >
      Random text
    </div>
  </div>
</div>
`;

exports[`Components : Notifications renders correctly 4`] = `
<div
  className="Notifications"
>
  <div
    className="Notifications-container"
  >
    <button
      className="Notifications-close-button"
      onClick={[Function]}
      type="button"
    >
      <svg
        className="Widget-icon"
        fill="currentColor"
        height="1em"
        stroke="currentColor"
        strokeWidth="0"
        style={
          Object {
            "color": undefined,
          }
        }
        viewBox="0 0 512 512"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M278.6 256l68.2-68.2c6.2-6.2 6.2-16.4 0-22.6-6.2-6.2-16.4-6.2-22.6 0L256 233.4l-68.2-68.2c-6.2-6.2-16.4-6.2-22.6 0-3.1 3.1-4.7 7.2-4.7 11.3 0 4.1 1.6 8.2 4.7 11.3l68.2 68.2-68.2 68.2c-3.1 3.1-4.7 7.2-4.7 11.3 0 4.1 1.6 8.2 4.7 11.3 6.2 6.2 16.4 6.2 22.6 0l68.2-68.2 68.2 68.2c6.2 6.2 16.4 6.2 22.6 0 6.2-6.2 6.2-16.4 0-22.6L278.6 256z"
        />
      </svg>
    </button>
    <div
      className="Notifications-content"
    />
  </div>
</div>
`;

exports[`Components : Notifications renders correctly 5`] = `
<div
  className="Notifications"
>
  <div
    className="Notifications-container"
  >
    <button
      className="Notifications-close-button"
      onClick={[Function]}
      type="button"
    >
      <svg
        className="Widget-icon"
        fill="currentColor"
        height="1em"
        stroke="currentColor"
        strokeWidth="0"
        style={
          Object {
            "color": undefined,
          }
        }
        viewBox="0 0 512 512"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M278.6 256l68.2-68.2c6.2-6.2 6.2-16.4 0-22.6-6.2-6.2-16.4-6.2-22.6 0L256 233.4l-68.2-68.2c-6.2-6.2-16.4-6.2-22.6 0-3.1 3.1-4.7 7.2-4.7 11.3 0 4.1 1.6 8.2 4.7 11.3l68.2 68.2-68.2 68.2c-3.1 3.1-4.7 7.2-4.7 11.3 0 4.1 1.6 8.2 4.7 11.3 6.2 6.2 16.4 6.2 22.6 0l68.2-68.2 68.2 68.2c6.2 6.2 16.4 6.2 22.6 0 6.2-6.2 6.2-16.4 0-22.6L278.6 256z"
        />
      </svg>
    </button>
    <div
      className="Notifications-content"
    />
  </div>
</div>
`;
